payments-core/src/main/java/com/stripe/android/paymentsheet/repositories/StripeIntentRepository.kt

```diff
@@ -29,27 +30,47 @@ internal sealed class StripeIntentRepository {
     class Api(
         private val stripeRepository: StripeRepository,
         private val requestOptions: ApiRequest.Options,
-        private val workContext: CoroutineContext
+        private val workContext: CoroutineContext,
+        private val locale: String? =
+            LocaleListCompat.getAdjustedDefault().takeUnless { it.isEmpty }?.get(0)?.toLanguageTag()
     ) : StripeIntentRepository() {
+        /**
+         * Tries to retrieve the StripeIntent with ordered Payment Methods, falling back to
+         * traditional GET if we don't have a locale or the call fails for any reason.
+         */
         override suspend fun get(clientSecret: ClientSecret) = withContext(workContext) {
             when (clientSecret) {
                 is PaymentIntentClientSecret -> {
-                    val paymentIntent = stripeRepository.retrievePaymentIntent(
-                        clientSecret.value,
-                        requestOptions,
-                        expandFields = listOf("payment_method")
-                    )
-                    requireNotNull(paymentIntent) {
+                    requireNotNull(
```

Comment:
What is the effect of this requireNotNull if the parsing fails or the network is down?

Reply:
If the parsing fails then the PI/SI is null, that's what this is checking for. If network is down an exception is thrown and this check is not executed.